contracts/utils/CurvedInterestRate.sol

```diff
@@ -55,8 +72,19 @@ contract CurvedInterestRate is ICurvedInterestRate {
         a = (_IR1 * 1e18) / ((_UR1 * b) / 1e18).exp();
     }
 
-    function calculateIR(uint256 _U, uint256 _IR1) public pure override returns (uint256 interestRate) {
-        (uint256 a, uint256 b) = getAB(_IR1);
-        return getInterestRate(_U, a, b);
+    function calculateIR(uint256 _U, uint256 _IR1) public view override returns (uint256 interestRate) {
+        CurveIRParams memory localParam = PARAMS[msg.sender];
+
+        if (localParam.IR_ABSOLUTE_MIN == 0) {
+            // those will be the defaults
+            localParam = PARAMS[address(0)];
+        }
+
+        (uint256 a, uint256 b) = getAB(_IR1, localParam.IR2, localParam.UR1, localParam.UR2, localParam.IR_MIN, localParam.IR_MAX);
+        return getInterestRate(_U, a, b, localParam.IR_MAX, localParam.IR_ABSOLUTE_MIN);
+    }
+
+    function updateParams(CurveIRParams calldata _curveIRParams, address owner) public onlyGuardian {
```

Comment:
I would prefer if this function is moved to LoanTokenLogicStandard as it is more intuitive and easier to maintain. Having another place to set guardian introduces more things to keep track of

Reply:
lets disscuss

Reply:
so after disscussion we decided its good to have it centralized for PL in the future. @Drypto13 please review again